packages/lodestar-validator/src/api/impl/rest/validator/validator.ts

```diff
@@ -66,21 +65,14 @@ export class RestValidatorApi implements IValidatorApi {
     return responseData.map((value) => this.config.types.Attestation.fromJson(value, {case: "snake"}));
   }
 
-  public async produceBlock(
-    slot: Slot,
-    proposerPubkey: BLSPubkey,
-    randaoReveal: Bytes96,
-    graffiti: string
-  ): Promise<BeaconBlock> {
-    const url = "/block";
+  public async produceBlock(slot: Slot, randaoReveal: Bytes96, graffiti: string): Promise<BeaconBlock> {
+    const url = `/blocks/${slot}`;
     const query = {
-      slot: slot,
-      proposer_pubkey: toHexString(proposerPubkey),
```

Comment:
so since we don't need to transfer pubkey to produce a block anymore, what's the plan to avoid DOS attack? I can write a script asking our beacon node to produce blocks multiple times to keep our node busy.

Reply:
how was pubkey preventing dos?

Reply:
hah so an attacker can easily find a pubkey and still can do the DOS attack.
digging into this, I think maybe we should validate randao_reveal when we produce a block?

it's not related to this PR anyway, just general concern.

Reply:
I mean producing block isn't performance intensive, limiting it to 1 req per second or something could suffice

Reply:
oh yeah that's why I don't see other clients are validating that

I'm fine with leaving it as is for now bc if we limit 1 req per second, an attacker can disturb our node and the real validators never has a chance to produce blocks. (if later on if we want to do it then probably we can base on client ip to ban attackers)

Reply:
sure, fyi rate limits are per ip not overall